source/common/config/watch_map_impl.h

```diff
@@ -0,0 +1,102 @@
+#pragma once
+
+#include <set>
+#include <string>
+#include <utility>
+
+#include "envoy/config/watch_map.h"
+
+#include "common/common/assert.h"
+#include "common/common/logger.h"
+
+#include "absl/container/flat_hash_map.h"
+#include "absl/container/flat_hash_set.h"
+
+namespace Envoy {
+namespace Config {
+
+// Manages "watches" of xDS resources. Several xDS callers might ask for a subscription to the same
```

Comment:
FWIW, I think these comments live best at the interface level. At least the first few paragraphs.

Reply:
Done, and revised/cleaned a bit.